web/data.py

```diff
@@ -43,22 +45,17 @@ def count_latest(df):
     return total
 
 
+# Data extractors
+
 def get_ph_confirmed():
     ph_conf = cache.get('ph_conf')
     if ph_conf is None:
         ph_url = 'https://ncovph.com/api/confirmed-cases'
         ph_conf = pd.read_json(request.urlopen(ph_url))
         ph_conf = ph_conf.drop('date_confirmed', axis=1)
-        regions, provinces, cities = [], [], []
-        for v in ph_conf['residence'].values:
-            if v is not None:
-                regions.append(v['region'])
-                provinces.append(v['province'])
-                cities.append(v['city'])
-            else:
-                regions.append(None)
-                provinces.append(None)
-                cities.append(None)
+        regions = [v['region'] if v is not None else None for v in ph_conf['residence'].values]
+        provinces = [v['province'] if v is not None else None for v in ph_conf['residence'].values]
+        cities = [v['city'] if v is not None else None for v in ph_conf['residence'].values]
```

Comment:
Good way to compact code; consider another loop for `regions`, `provinces`, and `cities`